File: app/src/main/java/com/fonteraro/martaco/cinemahub/utils/BindingAdapters.java
package com.fonteraro.martaco.cinemahub.utils;

import android.databinding.BindingAdapter;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

public abstract class BindingAdapters {

    @BindingAdapter({"bind:imageUrl"})
    public static void loadImage(ImageView view, String imageUrl) {
        Picasso.get()
                .load(imageUrl)
                .noPlaceholder()
                .noFade()
                .into(view);
    }
}

File: app/src/main/java/com/fonteraro/martaco/cinemahub/utils/BindingAdapters.kt
package com.fonteraro.martaco.cinemahub.utils

import androidx.appcompat.widget.AppCompatImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.fonteraro.martaco.cinemahub.R

@BindingAdapter("imageUrl")
fun loadImage(view: AppCompatImageView, imageUrl: String?) {
    if (imageUrl.isNullOrEmpty().not()) {
        Glide.with(view.context)
            .load(imageUrl)
            .dontAnimate()
            .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
            .placeholder(R.drawable.error_placeholder)
            .into(view)
    }
}
